asset-transfer-private-data/application-gateway-go/app.go

```diff
@@ -0,0 +1,358 @@
+	}
+
+	fmt.Println("\nAttempt DeleteAsset using non-owner organization")
+	err = deleteAsset(*contractOrg2, assetID2)
+	if err == nil {
+		doFail("DeleteAsset transaction succeeded when it was expected to fail")
+	}
+	fmt.Printf("*** Received expected error: %s\n", err.Error())
+
+	fmt.Println("\n~~~~~~~~~~~~~~~~ As Org1 Client ~~~~~~~~~~~~~~~~")
+
+	// Delete AssetID2 as Org1.
+	deleteAsset(*contractOrg1, assetID2)
+
+	// Trigger a purge of the private data for the asset.
+	// The previous delete is optional if purge is used.
+	purgeAsset(*contractOrg1, assetID2)
+}
+
+func createAssets(contract client.Contract) {
+	assetType := "ValuableAsset"
+
+	fmt.Printf("\n--> Submit Transaction: CreateAsset, ID: %s\n", assetID1)
+
+	type assetTransientInput struct {
+		ObjectType     string
+		AssetID        string
+		Color          string
+		Size           uint8
+		AppraisedValue uint16
+	}
+
+	asset1Data := assetTransientInput{
+		ObjectType:     assetType,
+		AssetID:        assetID1,
+		Color:          "green",
+		Size:           20,
+		AppraisedValue: 100,
+	}
+
+	if _, err := contract.Submit(
+		"CreateAsset",
+		client.WithTransient(transientData("asset_properties", asset1Data)),
+	); err != nil {
+		panic(err)
+	}
+
+	logTxCommitSuccess()
+	fmt.Printf("\n--> Submit Transaction: CreateAsset, ID: %s\n", assetID2)
+
+	asset2Data := assetTransientInput{
+		ObjectType:     assetType,
+		AssetID:        assetID2,
+		Color:          "blue",
+		Size:           35,
+		AppraisedValue: 727,
+	}
+
+	if _, err := contract.Submit(
+		"CreateAsset",
+		client.WithTransient(transientData("asset_properties", asset2Data)),
+	); err != nil {
+		panic(err)
+	}
+
+	logTxCommitSuccess()
+}
+
+func getAssetByRange(contract client.Contract) {
+	// GetAssetByRange returns assets on the ledger with ID in the range of startKey (inclusive) and endKey (exclusive).
+	fmt.Printf("\n--> Evaluate Transaction: GetAssetByRange from %s\n", org1PrivateCollectionName)
+
+	resultBytes, err := contract.EvaluateTransaction("GetAssetByRange", assetID1, fmt.Sprintf("asset%d", now.Unix()+2))
+	if err != nil {
+		panic(err)
+	}
+
+	result := formatJSON(resultBytes)
+	if result == "" {
+		doFail("Received empty query list for GetAssetByRange")
+	}
+	fmt.Printf("*** Result: %s\n", result)
+}
+
+func readAssetByID(contract client.Contract, assetID string) {
+	fmt.Printf("\n--> Evaluate Transaction: ReadAsset, ID: %s\n", assetID)
+
+	resultBytes, err := contract.EvaluateTransaction("ReadAsset", assetID)
+	if err != nil {
+		panic(err)
+	}
+
+	result := formatJSON(resultBytes)
+	if result == "" {
+		doFail("Received empty result for ReadAsset")
+	}
+	fmt.Printf("*** Result: %s\n", result)
+}
+
+func agreeToTransfer(contract client.Contract, assetID string) {
+	// Buyer from Org2 agrees to buy the asset.
+	// To purchase the asset, the buyer needs to agree to the same value as the asset owner.
+	dataForAgreement := struct {
+		AssetID        string
+		AppraisedValue int
+	}{assetID, 100}
```

Comment:
This seems to be the cause of the transfer failure. The agreement is hashed and is sensitive to the capitalization of the JSON property names, which the contract treats with an initial lower-case letter
```suggestion
	dataForAgreement := struct {
		AssetID        string `json:"assetID"`
		AppraisedValue int    `json:"appraisedValue"`
	}{assetID, 100}
```